NTUT-NPC/tat_gogogo
# [Feature] Add JWT support

# Description
as title add jwt support for auth management 
Fixes #13 

## Type of Change

- [ ] Bug fix (Issue / Bug fixed | Non-breaking)
- [x] Feature (New feature | Non-breaking)

*  add jwt middleware
*  add infrastructure layer for communicate between outside world and our app

# Test Cases
1. run app in local
2. send post request with form studentID and password
3. it will get token then copy it to [jwt](https://jwt.io/) and check out the result
4. send get request with query string `targetStudentID` `year` `semester` to get courses
5. send get request with query string `targetStudentID` to get curriculum semester

- [x] `/auth/curriculums/courses`
- [x] `/auth/curriculums/semesters`
